OptimizelySDK/Odp/LruCache.cs

```diff
@@ -118,9 +114,11 @@ public void Save(string key, T value)
                     {
                         var leastRecentlyUsedItem = _list.Last;
 
-                        var leastRecentlyUsedItemKey = (from cacheItem in _cache
-                            where cacheItem.Value == leastRecentlyUsedItem.Value
-                            select cacheItem.Key).FirstOrDefault();
+                        var leastRecentlyUsedItemKey =
+                            _cache.Where(
+                                    cacheItem => cacheItem.Value == leastRecentlyUsedItem.Value).
```

Comment:
Thanks for asking this question. It got me thinking more critically and then researching.  

---

The previous version is LINQ syntactic sugar on top of this new chained version to satisfy the linter. 🤷 

---

Regarding time complexity, as I understood/understand it, you're right the `.Where()` would be O(N).  The index `.FirstOrDefault()` should allow looping over the collection and stop on the first matching case, giving us O(1) [correct?] 

The `.Select()` will return the matching Key which is what's needed to remove the item from the cache.

cc @msohailhussain

Reply:
@jaeopt Thinking more on this: LINQ does have its own overhead. 

Should I change this to a standard loop?

Reply:
@mikechu-optimizely the current looping should be fixed since it'll slow down the replacement search significantly.
Wondering why `OrderedSet` solution was discarded? 
If it can't be used, we can also consider keeping `key` in the `ItemWrapper`. See https://github.com/optimizely/swift-sdk/blob/86a328f942da7955873727247ccd675d6b246f80/Sources/ODP/LruCache.swift#L84

Reply:
@jaeopt I've added the `Key` as you suggested to `ItemWrapper`.  This allowed me to remove the loop/LINQ. 👍 

I'm not unsure I see the `OrderedSet`.   Do you mean these lines that I'm replacing?
```
 var leastRecentlyUsedItemKey = (from cacheItem in _cache
    where cacheItem.Value == leastRecentlyUsedItem.Value
    select cacheItem.Key).FirstOrDefault();
```
if so, this LINQ expression is not different from the method chaining version. Happy to discuss on the scheduled meeting I sent for later today so we can close up this PR.

I like the newest solution and hope it meets your standard.  If you like the submission and Approve, we can cancel the meeting if you'd like.